### DISCUSSION TITLE
Strange console characters, only in docker

### DISCUSSION THREAD

**Discussion by mooserator** [itzg/docker-minecraft-server#739] [Q&A]
Hi,

Fairly new docker user here.  That said, I have worked in Java for many years and have reasonable ability with it.  I am hosting an RLCraft server for my son and his friend.  It is working well and they have access both inside and outside of our home.  The original build was running natively in Ubuntu, then I recently ported it to run within docker using itzg Minecraft image as the base.

My question is regarding use of the terminal or console screens in docker.  I usually access these through either docker attach or within Portainer where I can either attach to the session or view the log.  I prefer this over rcon use and am trying to understand why strange characters appear when using docker, but not when running natively on the server.  I think they are related to colors or styles, but am not certain.  

Can someone help explain this issue and if there is something I can do to change it?  While it is probably not a big deal, it is annoying and I would like to understand if I can fix it.  Appreciate any help or advice.

The Native console output when running in Ubuntu here is the goal.
![NativeConsoleOutput](https://user-images.githubusercontent.com/78229331/106321130-7e816800-6239-11eb-908b-f8ffb9c0df5c.JPG)

These others show the strange characters when docker is used, either from docker attach or within portainer.
Docker attach:
![DockerAttachConsoleOutput](https://user-images.githubusercontent.com/78229331/106321129-7de8d180-6239-11eb-9644-a16f96e4011f.JPG)
Portainer attach:
![PortainerAttachConsoleOutput](https://user-images.githubusercontent.com/78229331/106321131-7e816800-6239-11eb-9ae6-80d09cc7b997.JPG)
Portainer Log:
![PortainerLogConsoleOutput](https://user-images.githubusercontent.com/78229331/106321132-7e816800-6239-11eb-888d-e6f32865b462.JPG)

**Comment by itzg** [comment]
Those are [Minecraft console color codes](https://www.colorschemer.com/minecraft-color-codes/), so I would fully expect those to show up via some of the attach/log approaches when a modded server is being accessed.

Hopefully somebody else can suggest if/how to disable the color codes being used.

**Reply by mooserator** [reply]
Okay, thank you for the quick response :).  That gets me very close.  These seem to display "normally" when running the modded server directly in Ubuntu, just not within docker.  Would be sweet if it could be enabled within docker, but I don't know if that is realistic.

It appears like a --nojline option exists and should disable those codes.  Is it possible to inject or pass that argument through the docker run command?  I played with this using different -e options but nothing has triggered it yet.  Is the --nojline property mapped to an env variable?

https://github.com/itzg/docker-minecraft-server

**Reply by itzg** [reply]
To pass additional command-line arguments, you can pass them via the `EXTRA_ARGS` environment variable. (Apparently I need to document that option.)

**Reply by mooserator** [reply]
Excellent, thank you.  The container was removed, the "--nojline" argument was added to EXTRA_ARGS, and it was re-launched.  Either I am not using the argument right or it doesn't work for CurseForge RLCraft as the strange characters persist.  🤷‍♂️ Open to any other suggestions.  

```
docker run -d -t -i \
  -p 25565:25565 \
  --name mc \
  -e EULA=TRUE \
  -e TYPE=FORGE \
  -e VERSION=1.12.2 \
  -e FORGEVERSION=14.23.5.2854 \
  -e GENERIC_PACK=/modpacks/RLCraft+Server+Pack+1.12.2+-+Beta+v2.8.2.zip \
  -e GENERIC_PACK_STRIP_DIRS=1 \
  -e INIT_MEMORY=1G \
  -e MAX_MEMORY=4G \
  -e MAX_TICK_TIME=-1 \
  -e REPLACE_ENV_VARIABLES=TRUE \
  -e ONLINE_MODE=TRUE \
  -e ALLOW_FLIGHT=TRUE \
  -e DIFFICULTY=3 \
  -e VIEW_DISTANCE=6 \
  -e EXTRA_ARGS=--nojline \
  -v /opt/minecraft/RLCraft_docker:/data \
  -v /opt/minecraft/modpacks:/modpacks \
  itzg/minecraft-server 
```

**Reply by itzg** [reply]
This new option might help

https://github.com/itzg/docker-minecraft-server#interactive-and-color-console

**Reply by mooserator** [reply]
Interesting addition, thank you.  Will give it a try.  :)
